Homework-19

btorelli89 opened the pull request:
Homework19

btorelli89 committed:
Homework-19

Berichicko commented on src/test/java/homework19.java:
```diff
@@ -0,0 +1,42 @@
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class homework19 extends BaseTest {
```
**According to the Java style guide:** 

1. Class names should be nouns, in mixed cases with the first letter of each internal word capitalized. Interfaces names should also be capitalized just like class names.
2. Use whole words and must avoid acronyms and abbreviations.

public class **HomeWork19** extends BaseTest

Berichicko commented:
You got this problem, please correct it.

![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/cf28ee95-b775-43ab-9544-8897ebb74149)

**HINT**
Add   `<parameter name="BaseURL" value="https://qa.koel.app/"/>`

![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/03eb06b4-ce35-4f7a-8ad3-6efe57cf3e81)

btorelli89 commented:
Hello,

I sincerely appreciate your help. I changed the naming convention on HomeWork19. But the program is still having run errors. I also added the BaseURL parameter I had started to add yesterday. The checker says there are no syntax errors. Maybe it has something to do with the Xpath in choosePlaylist?

[cid:ab19934f-ae08-4135-b4b2-c6ceedc59a5e]
[cid:0de17e91-7a41-40ac-afda-c532c0327741]
________________________________
From: Aliaksandr (Alex) Zasinets ***@***.***>
Sent: Friday, June 14, 2024 12:09 AM
To: Test-Pro-Bootcamp/A59-QA-Automation-Framework ***@***.***>
Cc: Brittany Torelli ***@***.***>; Author ***@***.***>
Subject: Re: [Test-Pro-Bootcamp/A59-QA-Automation-Framework] Homework-19 (PR #73)


@Berichicko commented on this pull request.

________________________________

In src/test/java/homework19.java<https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/pull/73#discussion_r1639263315>:

> @@ -0,0 +1,42 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class homework19 extends BaseTest {


According to the Java style guide:

  1.  Class names should be nouns, in mixed cases with the first letter of each internal word capitalized. Interfaces names should also be capitalized just like class names.
  2.  Use whole words and must avoid acronyms and abbreviations.

public class HomeWork19 extends BaseTest

—
Reply to this email directly, view it on GitHub<https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/pull/73#pullrequestreview-2117481609>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BI5R44Y4W53JXEIHKPUUT5DZHJ3HPAVCNFSM6AAAAABJI4IJQ6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDCMJXGQ4DCNRQHE>.
You are receiving this because you authored the thread.

Berichicko commented:
> Hello, I sincerely appreciate your help. I changed the naming convention on HomeWork19. But the program is still having run errors. I also added the BaseURL parameter I had started to add yesterday. The checker says there are no syntax errors. Maybe it has something to do with the Xpath in choosePlaylist? [cid:ab19934f-ae08-4135-b4b2-c6ceedc59a5e] [cid:0de17e91-7a41-40ac-afda-c532c0327741]
> ________________________________ From: Aliaksandr (Alex) Zasinets ***@***.***> Sent: Friday, June 14, 2024 12:09 AM To: Test-Pro-Bootcamp/A59-QA-Automation-Framework ***@***.***> Cc: Brittany Torelli ***@***.***>; Author ***@***.***> Subject: Re: [Test-Pro-Bootcamp/A59-QA-Automation-Framework] Homework-19 (PR #73) @Berichicko commented on this pull request.
> ________________________________ In src/test/java/homework19.java<[#73 (comment)](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/pull/73#discussion_r1639263315)>:
> @@ -0,0 +1,42 @@
> +import org.openqa.selenium.By; +import org.openqa.selenium.WebElement; +import org.testng.Assert; +import org.testng.annotations.Test; + +public class homework19 extends BaseTest { According to the Java style guide: 1. Class names should be nouns, in mixed cases with the first letter of each internal word capitalized. Interfaces names should also be capitalized just like class names. 2. Use whole words and must avoid acronyms and abbreviations. public class HomeWork19 extends BaseTest — Reply to this email directly, view it on GitHub<[#73 (review)](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/pull/73#pullrequestreview-2117481609)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BI5R44Y4W53JXEIHKPUUT5DZHJ3HPAVCNFSM6AAAAABJI4IJQ6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDCMJXGQ4DCNRQHE>. You are receiving this because you authored the thread.

Hello

Please, re-submit homework for review

btorelli89 committed:
Homework-18

btorelli89 committed:
Homework-18

btorelli89 committed:
Homework-18

btorelli89 commented:
Hello,

I have resubmitted the HW with the suggested changes.

Kind regards,
Brittany
________________________________
From: Aliaksandr (Alex) Zasinets ***@***.***>
Sent: Friday, June 14, 2024 10:37 AM
To: Test-Pro-Bootcamp/A59-QA-Automation-Framework ***@***.***>
Cc: Brittany Torelli ***@***.***>; Author ***@***.***>
Subject: Re: [Test-Pro-Bootcamp/A59-QA-Automation-Framework] Homework-19 (PR #73)


Hello, I sincerely appreciate your help. I changed the naming convention on HomeWork19. But the program is still having run errors. I also added the BaseURL parameter I had started to add yesterday. The checker says there are no syntax errors. Maybe it has something to do with the Xpath in choosePlaylist? [cid:ab19934f-ae08-4135-b4b2-c6ceedc59a5e] [cid:0de17e91-7a41-40ac-afda-c532c0327741]
________________________________ From: Aliaksandr (Alex) Zasinets @.> Sent: Friday, June 14, 2024 12:09 AM To: Test-Pro-Bootcamp/A59-QA-Automation-Framework @.> Cc: Brittany Torelli @.>; Author @.> Subject: Re: [Test-Pro-Bootcamp/A59-QA-Automation-Framework] Homework-19 (PR #73<https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/pull/73>) @Berichicko<https://github.com/Berichicko> commented on this pull request.
________________________________ In src/test/java/homework19.java<#73 (comment)<https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/pull/73#discussion_r1639263315>>:
@@ -0,0 +1,42 @@
+import org.openqa.selenium.By; +import org.openqa.selenium.WebElement; +import org.testng.Assert; +import org.testng.annotations.Test; + +public class homework19 extends BaseTest { According to the Java style guide: 1. Class names should be nouns, in mixed cases with the first letter of each internal word capitalized. Interfaces names should also be capitalized just like class names. 2. Use whole words and must avoid acronyms and abbreviations. public class HomeWork19 extends BaseTest — Reply to this email directly, view it on GitHub<#73 (review)<https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/pull/73#pullrequestreview-2117481609>>, or unsubscribehttps://github.com/notifications/unsubscribe-auth/BI5R44Y4W53JXEIHKPUUT5DZHJ3HPAVCNFSM6AAAAABJI4IJQ6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDCMJXGQ4DCNRQHE. You are receiving this because you authored the thread.

Hello

Please, re-submit homework for review

—
Reply to this email directly, view it on GitHub<https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/pull/73#issuecomment-2168288493>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BI5R444H47OELSIMQO5IGXDZHME23AVCNFSM6AAAAABJI4IJQ6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNRYGI4DQNBZGM>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

Berichicko commented:
> Hello, I have resubmitted the HW with the suggested changes. Kind regards, Brittany
> […](#)
> ________________________________ From: Aliaksandr (Alex) Zasinets ***@***.***> Sent: Friday, June 14, 2024 10:37 AM To: Test-Pro-Bootcamp/A59-QA-Automation-Framework ***@***.***> Cc: Brittany Torelli ***@***.***>; Author ***@***.***> Subject: Re: [Test-Pro-Bootcamp/A59-QA-Automation-Framework] Homework-19 (PR #73) Hello, I sincerely appreciate your help. I changed the naming convention on HomeWork19. But the program is still having run errors. I also added the BaseURL parameter I had started to add yesterday. The checker says there are no syntax errors. Maybe it has something to do with the Xpath in choosePlaylist? [cid:ab19934f-ae08-4135-b4b2-c6ceedc59a5e] [cid:0de17e91-7a41-40ac-afda-c532c0327741]
> ________________________________ From: Aliaksandr (Alex) Zasinets @.> Sent: Friday, June 14, 2024 12:09 AM To: Test-Pro-Bootcamp/A59-QA-Automation-Framework @.> Cc: Brittany Torelli @.>; Author @.> Subject: Re: [Test-Pro-Bootcamp/A59-QA-Automation-Framework] Homework-19 (PR #73<#73>) @Berichicko<https://github.com/Berichicko> commented on this pull request.
> ________________________________ In src/test/java/homework19.java<#73 (comment)<[#73 (comment)](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/pull/73#discussion_r1639263315)>>: @@ -0,0 +1,42 @@ +import org.openqa.selenium.By; +import org.openqa.selenium.WebElement; +import org.testng.Assert; +import org.testng.annotations.Test; + +public class homework19 extends BaseTest { According to the Java style guide: 1. Class names should be nouns, in mixed cases with the first letter of each internal word capitalized. Interfaces names should also be capitalized just like class names. 2. Use whole words and must avoid acronyms and abbreviations. public class HomeWork19 extends BaseTest — Reply to this email directly, view it on GitHub<#73 (review)<[#73 (review)](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/pull/73#pullrequestreview-2117481609)>>, or unsubscribehttps://github.com/notifications/unsubscribe-auth/BI5R44Y4W53JXEIHKPUUT5DZHJ3HPAVCNFSM6AAAAABJI4IJQ6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDCMJXGQ4DCNRQHE. You are receiving this because you authored the thread. Hello Please, re-submit homework for review — Reply to this email directly, view it on GitHub<[#73 (comment)](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/pull/73#issuecomment-2168288493)>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BI5R444H47OELSIMQO5IGXDZHME23AVCNFSM6AAAAABJI4IJQ6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCNRYGI4DQNBZGM>. You are receiving this because you authored the thread.Message ID: ***@***.***>

Hi dear Brittany

I will help you! Don't worry!

You got this problem!

![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/6e5b02b7-8b31-4184-839e-6e7d45de1a30)

**HINT**

**Step 1**

**NoSuchElementException**  

1. Verify the locator: As a first step, you ought to find out what kind of issue you are dealing with. Is the element still there and can't be found, or is it no longer present on the page? To do this, you can setup Selenium to not to close the browser after the test failure, and run the test. Once you get the exception, you can inspect the page that is open in the browser. Double-check that the locator being used is correct and up-to-date. Inspect the HTML source code to confirm that the locator uniquely identifies the target element on the page.

2. Wait for the element: Implement implicit (not recommended unless necessary) or explicit (recommended) waits to give the element time to load before attempting to interact with it. This approach helps to ensure that Selenium does not attempt to find the element before it has been rendered on the page.


**Step 2**

**You can use this extension for easy locator search.**

 Ranorex Selocity

Ranorex Selocity is a free extension for Chrome DevTools. It allows you to create unique selectors through a smart selector generator. Feedback is created, allowing you to locate your web page's elements in the most secure manner. https://www.ranorex.com/selocity/browser-extension/

For example 

![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/d7f9b33c-8692-4fd8-b680-fb0da31ffd7b)

**Xpath cheatsheet** https://devhints.io/xpath 

**Incorrect:**

![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/59fb5cc3-30f9-4510-be0d-1d352ddc23f8)

**Correctly:**

![Screenshot_2](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/7ebfbcba-281b-4191-bcac-e0b35f4b403f)

![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/ea534f92-ab9e-4405-a865-1d240c7a54cf)

btorelli89 committed:
Homework-18

btorelli89 committed:
Homework19

I fixed the errors. It deletes a playlist and gives me the message. Let me know if there is anything else I need to fix.

Berichicko commented:
Hello,

I have checked your homework 19, and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, [aliaksandr@testpro.io](mailto:aliaksandr@testpro.io))!

![Thumbs-up-hand-with-crown-illustration-on-transparent-background-PNG](https://github.com/Tootienahs/A59-QA-Automation-Framework/assets/104761482/20861dce-f8e7-4435-9b92-de6436933d0b)

![Quote](https://github-readme-quotes-bay.vercel.app/quote?theme=dark&layout=socrates)


![image](https://github.com/Test-Pro-Bootcamp/A59-QA-Automation-Framework/assets/104761482/e0163419-855a-4fd9-bce3-d05ab057ecb7)
